pageserver/src/tenant.rs

```diff
@@ -137,6 +138,7 @@ pub struct Tenant {
 
     /// Cached logical sizes updated updated on each [`Tenant::gather_size_inputs`].
     cached_logical_sizes: tokio::sync::Mutex<HashMap<(TimelineId, Lsn), u64>>,
+    cached_synthetic_tenant_size: Arc<AtomicU64>,
```

Comment:
I don't think this is required, since the inputs will be cached by each call to `gather_size_inputs`, so if no lsn has advanced two calls, nothing will need to be recalculated.

The http API is for adhoc testing usage anyways, so it's not supposed to be called for any other reason.

If we want to cache this, then we should make it a `Mutex<Option<(Instant, u64)>>`, to fill it as:

```rust
let expires_in = std::time::Duration::from_secs(60 * 15);

{
  let g = cached_synthetic_tenant_size.lock().unwrap();
  if let Some((expires_at, value)) = g.as_ref() {
    if expires_at.elapsed() < expires_in {
      return *value;
    }
  }
}

let model = tenant.gather_size_inputs().await?;
let size = model.calculate()?;

{
  let now = std::time::Instant::now();
  let mut g = cached_synthetic_tenant_size.lock().unwrap();
  *g = Some((now + expires_in, size));
}
```

But I doubt it will be necessary, only if someone is doing requests against many of the tenants over the HTTP api.